In [2]:
library(tidyverse)
library(dplyr)
library(ggplot2)

players  <- read.csv("players.csv")
sessions <- read.csv("sessions.csv")

head(players)
head(sessions)
summary(players)
summary(sessions)

,experience,subscribe,hashedEmail,played_hours,name,gender,Age
,<chr>,<lgl>,<chr>,<dbl>,<chr>,<chr>,<int>
1,Pro,TRUE,f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d,30.3,Morgan,Male,9
2,Veteran,TRUE,f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9,3.8,Christian,Male,17
3,Veteran,FALSE,b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28,0.0,Blake,Male,17
4,Amateur,TRUE,23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5,0.7,Flora,Female,21
5,Regular,TRUE,7dc01f10bf20671ecfccdac23812b1b415acd42c2147cb0af4d48fcce2420f3e,0.1,Kylie,Male,21
6,Amateur,TRUE,f58aad5996a435f16b0284a3b267f973f9af99e7a89bee0430055a44fa92f977,0.0,Adrian,Female,17


,hashedEmail,start_time,end_time,original_start_time,original_end_time
,<chr>,<chr>,<chr>,<dbl>,<dbl>
1,bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf,30/06/2024 18:12,30/06/2024 18:24,1.71977e+12,1.71977e+12
2,36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686,17/06/2024 23:33,17/06/2024 23:46,1.71867e+12,1.71867e+12
3,f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc,25/07/2024 17:34,25/07/2024 17:57,1.72193e+12,1.72193e+12
4,bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf,25/07/2024 03:22,25/07/2024 03:58,1.72188e+12,1.72188e+12
5,36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686,25/05/2024 16:01,25/05/2024 16:12,1.71665e+12,1.71665e+12
6,bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf,23/06/2024 15:08,23/06/2024 17:10,1.71916e+12,1.71916e+12


  experience        subscribe       hashedEmail         played_hours    
 Length:196         Mode :logical   Length:196         Min.   :  0.000  
 Class :character   FALSE:52        Class :character   1st Qu.:  0.000  
 Mode  :character   TRUE :144       Mode  :character   Median :  0.100  
                                                       Mean   :  5.846  
                                                       3rd Qu.:  0.600  
                                                       Max.   :223.100  
                                                                        
     name              gender               Age       
 Length:196         Length:196         Min.   : 9.00  
 Class :character   Class :character   1st Qu.:17.00  
 Mode  :character   Mode  :character   Median :19.00  
                                       Mean   :21.14  
                                       3rd Qu.:22.75  
                                       Max.   :58.00  
                               

 hashedEmail         start_time          end_time         original_start_time
 Length:1535        Length:1535        Length:1535        Min.   :1.712e+12  
 Class :character   Class :character   Class :character   1st Qu.:1.716e+12  
 Mode  :character   Mode  :character   Mode  :character   Median :1.719e+12  
                                                          Mean   :1.719e+12  
                                                          3rd Qu.:1.722e+12  
                                                          Max.   :1.727e+12  
                                                                             
 original_end_time  
 Min.   :1.712e+12  
 1st Qu.:1.716e+12  
 Median :1.719e+12  
 Mean   :1.719e+12  
 3rd Qu.:1.722e+12  
 Max.   :1.727e+12  
 NA's   :2          

## (1) Data Description

### Dataset Overview

| File | Rows (Observations) | Columns (Variables) | Description |
|------|----------------------|---------------------|--------------|
| `players.csv` | 196 | 7 | Player-level demographics and overall play summary |
| `sessions.csv` | 1,535 | 5 | Session-level time data (start and end timestamps) |

---

### Variables in Each File

| Dataset | Variable | Type | Description | Summary |
|----------|-----------|------|-------------|----------|
| **players.csv** | `experience` | Character | Player's experience level (`Beginner`, `Regular`, `Pro`, etc.) | 196 obs |
|  | `subscribe` | Logical | Whether player subscribed to newsletter (`TRUE`/`FALSE`) | 144 TRUE, 52 FALSE |
|  | `hashedEmail` | Character | Hashed email address (unique ID) | Unique across players |
|  | `played_hours` | Double | Total hours played | Mean = 5.85, Max = 223.10 |
|  | `name` | Character | Player name | Text |
|  | `gender` | Character | Player gender (`Male`, `Female`, `Other`) | 196 obs |
|  | `Age` | Integer | Player age in years | Mean = 21.14, Range = 9–58 |
| **sessions.csv** | `hashedEmail` | Character | Player ID linking sessions to players | 1,535 obs |
|  | `start_time` | Character | Session start time (ms since epoch) | Requires conversion |
|  | `end_time` | Character | Session end time (ms since epoch) | Requires conversion |
|  | `original_start_time` | Character | Raw start timestamp | NA’s = 2 |
|  | `original_end_time` | Character | Raw end timestamp | NA’s = 2 |

---

### Key Notes

- Both datasets link via **`hashedEmail`**, which uniquely identifies each player.  
- **Timestamps** are stored in **milliseconds**, requiring conversion to readable time units for analysis.  
- The datasets differ in **granularity**: `players.csv` summarizes each player, while `sessions.csv` records every individual session.

### Data Direct and Potential Issues
**Observable Issues**
- **Missing values:** Some missing timestamps or ages (`NA`).
- **Outliers:** Duration times show skewed distribution; few players exceed 4 hours per session.
- **Formatting:** Time columns stored as numeric/character (ms), requiring conversion to minutes.

**Potential Hidden Issues**
- **Sampling bias:** Active players may be overrepresented.
- **Time-zone uncertainty:** Logs may mix multiple time zones.
- **Device/network:** One player could appear multiple times from reconnects.

### (2) Question
Which kinds of players are most likely to contribute a large amount of data (i.e., play for a long time)?

### Specific Predictive Question
Can **early play behaviour** — such as the **number of sessions** and **average session duration** in the first 7 days — predict whether a player becomes a **high contributor** (top 25% of total play time)?

### Response Variable (Y)
`high_contributor` — Binary indicator (1 = Top 25% of total minutes played)

### Explanatory Variables (X)
- `sessions_n_early` – Number of sessions in first 7 days  
- `avg_duration_early` – Average duration of early sessions (minutes)

### How the Data Help
- `sessions.csv` → used to compute per-player total and early play time  
- `players.csv` → demographic context (gender, experience, etc.)  
Merging both allows us to explore relationships between **early engagement** and **overall contribution**.
